io.greenfire.Greenery.json

```diff
@@ -0,0 +1,77 @@
+{
+    "app-id": "io.greenfire.Greenery",
+    "base": "org.electronjs.Electron2.BaseApp",
+    "base-version": "23.08",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "greenery",
+    "separate-locales": false,
+    "tags": [
+        "proprietary"
+    ],
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--device=dri"
+    ],
+    "modules": [
+        {
+            "name": "greenery",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 io.greenfire.Greenery.desktop /app/share/applications/io.greenfire.Greenery.desktop",
+                "install -Dm644 io.greenfire.Greenery.appdata.xml /app/share/appdata/io.greenfire.Greenery.appdata.xml",
+                "install -Dm644 -t /app/share/icons/hicolor/scalable/apps io.greenfire.Greenery.svg",
+                "install -D apply_extra /app/bin/apply_extra",
+                "install -D greenery /app/bin/greenery"
+            ],
+            "sources": [
+                {
+                    "type": "script",
+                    "dest-filename": "apply_extra",
+                    "commands": [
+                        "unzip greenery.zip > /dev/null",
+                        "mv Greenery-linux-x64/* .",
+                        "rm -r Greenery-linux-x64 *.zip"
+                    ]
+                },
+                {
+                    "type": "script",
+                    "dest-filename": "greenery",
+                    "commands": [
+                        "export TMPDIR=\"$XDG_RUNTIME_DIR/app/$FLATPAK_ID\"",
+                        "exec zypak-wrapper /app/extra/greenery \"$@\""
+                    ]
+                },
+                {
+                    "type": "extra-data",
```

Comment:
I am one of the developers for Greenery.  I tried using the archive type but got these errors when trying to run Greenery.  I have pushed my changes to the PR.  I was able to run the app with the sandbox wrapper by using /app/extra/greenery when using extra-data.  I'm not sure how to get it to run when using: archive

[704 preload-host-spawn-strategy] Warning: waitpid override ignores groups
[704 preload-host-spawn-strategy] Warning: waitpid override ignores groups
[2 preload-host-spawn-strategy] Warning: waitpid override ignores groups
[2 preload-host-spawn-strategy] Warning: waitpid override ignores groups
[2 zypak-helper] Using spawn strategy test 1 as set by environment
[2 zypak-helper] /app/bin/greenery is not an ELF file
[2 zypak-helper] (it appears to be a shell script?)
[2 zypak-helper] WARNING: supplied target /app/bin/greenery does not look like a valid Chromium binary!
[2 zypak-helper] Zypak needs to be called directly on the executable *binary*, not any wrappers.

Reply:
What does `/app/bin/greenery` look like? It's like it says, if it is a wrapper it won't work, you have to point it to the actual binary.

Reply:
How is it referencing itself? you have `exec zypak-wrapper /app/bin/greenery` <--- this can't be right?

Reply:
I have pushed up my latest code.  I set the type to archive like this:

                    "type": "archive",
                    "url": "https://github.com/GreenfireInc/Releases.Greenery/releases/download/v0.9.7/Greenery-linux-x64-0.9.7.zip",
                    "sha256": "b230efd545589effebdade8f20162801b71733f2f267c1f9a2ad7e285bc042fd", 
                    "dest": "main",
                    "dest-filename": "greenery.zip",
                    
                   
I can see that setting "dest" to "main" decompresses the greenery binary and associated files into the local directory:

./.flatpak-builder/build/greenery-1/main

I can run the greenery binary from there and it launches the app without issue.  However when the app continues to build it never moves over the binary and associated files to /app/main/ so the following script call fails:

                    "type": "script",
                    "dest-filename": "greenery",
                    "commands": [
                        "export TMPDIR=\"$XDG_RUNTIME_DIR/app/$FLATPAK_ID\"",
                        "exec zypak-wrapper /app/main/greenery \"$@\""
                    ]

So I need to figure out how to copy these files over so that they appear here:

~/.local/share/flatpak/app/io.greenfire.Greenery/current/active/files/main

Reply:
> However when the app continues to build it never moves over the binary and associated files to /app/main/ so the following script call fails:

You think it will move it for you? flatpak-builder runs command as told. It doesn't try to guess. So you have to add the commands to do so.

Reply:
I used a post-install section to move the files from /main (includes greenery binary) to /app/main/ and then the app was able to run.  Let me do some more testing and I will push my updates.

Is it ok to use post-install for moving the files and creating the main directory?  This is what the code looks like:

"post-install": [
                "mkdir -p /app/main/",
                "mv main/* /app/main/"
            ]

Reply:
I have moved the mkdir and mv for /main into the build-commands section, thus removing the post-install section.  The binary now gets moved into /app/main/greenery and the script command is able to run greenery with the wrapper:

"exec zypak-wrapper /app/main/greenery \"$@\""

This should build and run ok with the updated archive type.